templates/base/packages/nextjs/components/ScaffoldEthAppWithProviders.tsx.template.mjs

```diff
@@ -15,9 +15,21 @@ const contents = ({ preContent, globalClassNames, extraProviders, overrideProvid
       ? overrideProviders[0]
       : { ...defaultProviders, ...(extraProviders[0] || {}) };
 
-  const providers = Object.entries(providersObject).map(([providerName, props]) =>
-    `$$createProvider(${providerName}, ${JSON.stringify(props)})$$`
-  );
+  const providerEntries = Object.entries(providersObject);
+  const providerOpeningTags = providerEntries.map(([providerName, props], index) => {
+    const propAssignments = Object.entries(props).map(([propName, propValue]) => 
+      `${propName}={${stringify(propValue)}}`
+    ).join(' ');
+    const openingIndentation = index === 0 ? '' : '    ' + '  '.repeat(index);
+    return `${openingIndentation}<${providerName} ${propAssignments}>`;
+  }).join('\n');
+
+  const contentIndentation = '  '.repeat(providerEntries.length);
+
+  const providerClosingTags = providerEntries.reverse().map(([providerName], index) => {
+    const closingIndentation = (index === 0 ? '' : '    ') + '  '.repeat(providerEntries.length - index - 1);
+    return `${closingIndentation}</${providerName}>`;
+  }).join('\n');
```

Comment:
Can we remove the indentation logic? Since anywhich ways it will be formatted by prettier so lets remove that logic : 

```ts
  const providerEntries = Object.entries(providersObject);
  const providerOpeningTags = providerEntries.map(([providerName, props]) => {
    const propAssignments = Object.entries(props).map(([propName, propValue]) => 
      `${propName}={${stringify(propValue)}}`
    ).join(' ');
    return `<${providerName} ${propAssignments}>`;
  }).join('\n');

  const providerClosingTags = providerEntries.reverse().map(([providerName]) => {
    return `</${providerName}>`;
  }).join('\n');

```